### PR TITLE
Move work queue status trigger from cloud

### PR DESCRIPTION
**Pull request by pleek91**

### CODE REVIEW

File: src/automations/maps/workQueueStatusTrigger.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    }
+  }
+
+  throw new Error(`Unknown work queue status events: ${statusEvents}`)
+}
+
+function mapWorkQueueStatusToEvent(status: WorkPoolQueueStatus): WorkQueueStatusEvent {
+  switch (status) {
+    case 'not_ready':
+      return 'prefect.work-queue.not-ready'
```
Comment (znicholasbrown) [inline src/automations/maps/workQueueStatusTrigger.ts:98]: I forget, does this need a `not-ready` case as well?

**Review by znicholasbrown — APPROVED** [review]

File: src/automations/maps/workQueueStatusTrigger.ts (hunk 1 above)
Comment (pleek91) [reply src/automations/maps/workQueueStatusTrigger.ts:98]: I believe this is correct but let me verify

Comment (pleek91) [reply src/automations/maps/workQueueStatusTrigger.ts:98]: Verified with @collincchoy this is correct 👍

Comment (znicholasbrown) [reply src/automations/maps/workQueueStatusTrigger.ts:98]: Great, thank you 🙏🏻

**Pull request merged**